models/message.go

```diff
@@ -0,0 +1,51 @@
+package models
+
+import (
+	"bytes"
+	"encoding/base32"
+
+	"github.com/pborman/uuid"
+	"github.com/pkg/errors"
+)
+
+type NewMessageResponse struct {
+	Id      string `json:"id"`
+	Message string `json:"message"`
+}
+
+type Message struct {
+	Id      string `json:"id"`
+	Subject string `json:"subject"`
+	Text    string `json:"text"`
+	From    string `json:"from"`
+	To      string `json:"recipients"`
+	Status  string `json:"status"`
+}
+
+type QueueMessage struct {
+	Id   string `json:"id"`
+	Type string `json:"type"`
+}
+
+// After marshaling from JSON, call this method to validate the object is intact
+func (self *Message) Validate() error {
+
+	if err := ValidEmail(self.From); err != nil {
+		return errors.Wrap(err, "From")
+	}
+
+	if err := ValidEmail(self.To); err != nil {
+		return errors.Wrap(err, "To")
+	}
+
+	return nil
+}
+
+func NewId() string {
```

Comment:
By using "hash/fnv" I meant hashing some random value.

Disclaimer: I used fnv to make IDs by hashing some unique meta-data of an item. :)